src/input/storage_transpose.cc

```diff
@@ -112,7 +112,8 @@ StorageBase * StorageTranspose::modify_storage(const Type::TypeBase *target_type
 	ASSERT( target_arec != NULL, "Incompatible type of target type. Must be AbstractRecord!\n");
 	ASSERT( typeid(*source_storage) == typeid(StorageArray), "Incompatible type of storage. Must be Array!\n");
 
-	unsigned int descendant_index = (unsigned int)(source_storage->get_item(0)->get_int());
+	string descendant_name = source_storage->get_item(0)->get_string();
```

Comment:
Try to use directly get_descendant(string) 
possibly we can remove Record::get_descendant(int)